src/server/db_slice.h

```diff
@@ -384,6 +388,16 @@ class DbSlice {
 
   // Registered by shard indices on when first document index is created.
   DocDeletionCallback doc_del_cb_;
+
+  struct Hash {
+    size_t operator()(const facade::Connection::WeakRef& c) const {
+      return std::hash<uint32_t>()(c.GetClientId());
+    }
+  };
+
+  // the table that maps keys to the clients that are tracking them.
+  absl::flat_hash_map<std::string, absl::flat_hash_set<facade::Connection::WeakRef, Hash>>
```

Comment:
what is the number of keys in this map? relay spoke about 1 milion keys tracked right?

Reply:
we should add tracking to the memory usage here

Reply:
@theyueli ping on this one